### ISSUE TITLE
28170362: Xcode-beta (8S201h): Xcode extensions can't make the compiler emit warnings

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#15847]
#### Description

Summary:
Xcode 8 extensions are not able to force the compiler to show a warning at a certain line. 

Steps to Reproduce:
- Create an Xcode 8 extension
- Have a situation where you’d prefer to show a warning rather than changing the text directly
- Have no way to do that

Expected Results:
- Would like to be able to emit warnings via the compiler for things which need to be addressed

Actual Results:
- Can’t. 

Regression:
(Not a regression, this is a new feature)

Notes:
Basically this would be really helpful for any kind of linting extension, because it brings greater awareness to the user of when they’ve violated a particular linter’s spec. It could also be used to emit warnings for possible non-localized strings or other stringly-typed code likely to fail because we are all awful spellers.
## 

Product Version: Xcode-beta (8S201h)
Created: 2016-09-06T14:30:08.251420
Originated: 2016-09-06T15:29:00
Open Radar Link: http://www.openradar.me/28170362
